examples/involvingstates/example_pgm_pbm.py

```diff
@@ -0,0 +1,242 @@
+from toqito.states import trine
+from toqito.state_opt import state_distinguishability, state_exclusion
+
+
+def calculate_success_prob(
+    states: list[np.ndarray],
+    probs: list[float],
+    povm_operators: list[np.ndarray],
+) -> float:
+    """Calculate the success probability Σ pᵢ Tr(ρᵢ Mᵢ).
+
+    This helper is robust to `states` being either state vectors or density matrices.
+    """
+    success_prob = 0
+    num_states = len(states)
+    for i in range(num_states):
+        state = states[i]
+        op = povm_operators[i]
+        # Check if input is a vector (pure state) or matrix (density matrix)
+        if state.ndim == 1 or (state.ndim == 2 and min(state.shape) == 1):
+            # It's a vector (or column/row vector)
+            state_vec = state.flatten()
+            prob_i = state_vec.conj().T @ op @ state_vec
+        else:
+            # It's a density matrix
+            prob_i = np.trace(op @ state)
+        success_prob += probs[i] * prob_i
+    return np.real(success_prob)
+
+
+# 1. Define the states and probabilities.
+state_vectors = trine()
+k = len(state_vectors)
+probs = [1 / k] * k
+
+print(f"Analyzing k={k} trine states with uniform probability.")
+
+# 2. Compute the optimal benchmark values.
+p_best, _ = state_distinguishability(state_vectors, probs)
+p_worst, _ = state_exclusion(state_vectors, probs)
+
+print(f"\nOptimal Benchmarks:")
+print(f"  P_Best  = {p_best:.4f} (Max discrimination probability)")
+print(f"  P_Worst = {p_worst:.4f} (Min discrimination probability)")
+
+# 3. Compute the PGM and PBM operators.
+pgm_operators = pretty_good_measurement(state_vectors, probs)
+pbm_operators = pretty_bad_measurement(state_vectors, probs)
+
+# 4. Calculate PGM and PBM success probabilities.
+p_pgm = calculate_success_prob(state_vectors, probs, pgm_operators)
+p_pbm = calculate_success_prob(state_vectors, probs, pbm_operators)
+
+print(f"\nHeuristic Measurements:")
+print(f"  P_PGM = {p_pgm:.4f}")
+print(f"  P_PBM = {p_pbm:.4f}")
+
+# 5. Verify the core relationship and the hierarchy.
+relation_lhs = p_pgm + (k - 1) * p_pbm
+print(f"\nVerifying P_PGM + (k-1)*P_PBM = 1:")
+print(f"  {p_pgm:.4f} + ({k - 1})*{p_pbm:.4f} = {relation_lhs:.4f} -> {np.isclose(relation_lhs, 1)}")
+
+print("\nVerifying hierarchy (P_Best >= P_PGM >= 1/k >= P_PBM >= P_Worst):")
+print(f"  {p_best:.4f} >= {p_pgm:.4f}  ?  {p_best >= p_pgm or np.isclose(p_best, p_pgm)}")
+print(f"  {p_pgm:.4f} >= {1 / k:.4f}  ?  {p_pgm >= 1 / k or np.isclose(p_pgm, 1 / k)}")
+print(f"  {1 / k:.4f} >= {p_pbm:.4f}  ?  {1 / k >= p_pbm or np.isclose(1 / k, p_pbm)}")
+print(f"  {p_pbm:.4f} >= {p_worst:.4f}?  {p_pbm >= p_worst or np.isclose(p_pbm, p_worst)}")
+
+
+# %%
+# The results perfectly match those presented in Figure 3 of the paper. For
+# the trine states, the PGM is optimal (:math:`P_{\text{PGM}} = P_{\text{Best}} = 2/3`)
+# and the worst possible measurement gives zero success probability
+# (:math:`P_{\text{Worst}} = 0`).
```

Comment:
move this to line 144 after the first set of print statements for the results. It is easier to read the results this way.